### ISSUE TITLE
default color scale should be identity

### ISSUE THREAD

**Issue by Fil** [observablehq/plot#744]
When no color channel is activated, the color scale defaults to linear-turbo, which is not a problem in the chart (since it's not used), but then shows as a turbo color *legend* coming out of nowhere.
<img width="466" alt="image" src="https://user-images.githubusercontent.com/7001/153256985-165d2711-45c9-495d-ace0-b9bb054c20bb.png">

(reported by @yurivish)

**Comment by mbostock** [comment]
If there are no color channels, shouldn’t the color scale be null, and then no associated color legend?

Or is this code saying `color: {legend: true}`, which means that there _is_ a color scale, even though it’s not actually being used? If that’s the case, I think this is the correct behavior. I don’t see why we would default to the identity scale over a linear scale for color. The x and y scales also default to linear.

**Comment by Fil** [comment]
Yes it's when you specify constant colors everywhere then request a legend. Plot doesn't know what the colors are —which is understandable since they haven't gone through a scale—, but the default color then comes as a surprise.

But you're right the color scale should probably not be identity but undefined, like in
> Plot.dotX([1, 2]).plot().scale("y") // undefined

**Comment by mbostock** [comment]
I’m saying I think the current behavior is correct. When you say `color: {}` you’re saying that the plot has a color scale (with all the defaults), even though that color scale isn’t used anywhere. When you add `color: {legend: true}` you’re further saying that you want a legend to be generated for that color scale. If you don’t want a color scale, when don’t say `color: {legend: true}`.

Perhaps if you say `color: {}` but don’t use it anywhere we could just ignore it, but my intent there was to respect the user’s intention in declaring the scale in the first place. And again, I think it’s more appropriate for the color scale to default to linear rather than identity, but I guess we could add a special case if there are no channels associated with the scale if you feel this is important?

**Comment by Fil** [comment]
Definitely not crucial, but the motivating example seems compelling.

The author has put in constant colors in the chart, then asked for a legend, maybe thinking about these colors (or left `{legend: true}` from an earlier iteration of the code), and got this surprising legend which wasn't connected in any way with the chart.

If the same color names had been specified as a channel—e.g., `dot(…{fill: ["blue", "red"] })`—, then the scale would have defaulted to "identity" and the legend would have been undefined.

There can be cases where the author's intent is indeed to create a color scale without a channel (for example, I did this when experimenting with the hexbin layout to set the colors of the bins), but I think one would then use at least `color: {type:"linear"}` or `color: {domain: [0, 10] }`?

I've updated the title to reflect the more limited scope of the question, and will think more about this.

**Comment by mbostock** [comment]
Something to think about: what would you expect this to return?

```js
Plot.dotX([], {fill: "foo"}).plot().scale("color")
```

Here the color scale is in use, but there aren’t any associated values (because the one mark that applies the color scale has empty data). Would you expect the color scale here to default to an identity scale or a linear scale? My take is that it should be a linear scale because that is the most common case. Or at least that’s what Plot has standardized on as the default scale type if all other hints fail, with the expectation that channels are more commonly used to encode data rather than express literal values.

https://github.com/observablehq/plot/blob/8bffb4bdaa3437b96766f32bf86f143226cf3eaf/src/scales.js#L251

**Comment by Fil** [comment]
In that case I think it should be left undecided, and default to null (or maybe type:"none" if we want a type—I used type:"identity" because it's as good as null in terms of legend, but I realize how it can be confusing).
